### ISSUE TITLE
Rule: `no-undefined-return-in-useStoreMap`

### ISSUE THREAD

**Issue by AlexandrHoroshih** [effector/eslint-plugin#115]
**Propolsal**
Rule, that makes return of `undefined` forbidden. If user actually want to use skip-update behavoiur, one can do explicit `return undefined` and suppress it with `eslint-disable-next-line`

Bad code example:
```ts
const item = useStoreMap({
  store: $kv,
  keys: [itemId],
  fn: (kv, [id]) => kv[id]
})
```

Good code example:
```ts
const item = useStoreMap({
  store: $kv,
  keys: [itemId],
  fn: (kv, [id]) => kv[id] ?? null // fallback to null
})
```

**Background**
`effector-react`'s hook `useStoreMap` historically has a very specific behavior for the case where `fn` returns `undefined` - it just skips update, just like original effector's store does

But in case of React it can lead to confusing and unwanted rendering behavior and can happen implicitly

Typical case is like this:
```ts
const item = useStoreMap({
  store: $kv,
  keys: [itemId],
  fn: (kv, [id]) => kv[id]
})

if (!item) return null;

return <Item {...item} />
```
Here, if `$kv` does not contain item with provided `itemId`, `fn` will return undefined and there will be no rerender. For the user it will look like if React did not updated after `itemId` change and previous render is "stuck" in the tree

This behavior been there for a long time and there may be cases where people actually depend on it, so it cant be just removed either, so the rule is needed first

**Comment by igorkamyshev** [comment]
As I understand [it is the same issue](https://github.com/effector/eslint-plugin/issues/10). So, let's merge it.
